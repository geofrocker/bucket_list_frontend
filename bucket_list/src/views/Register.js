import React, {Component} from 'react';
import {Link} from 'react-router-dom';


class Register extends Component{
    render(){
        return (
            <div>
                <div className="auth">
                    <div className="auth-container">
                        <div className="card">
                            <header className="auth-header">
                                <h1 className="auth-title">
                                    BUCKET-LIST
                                </h1>
                            </header>
                            <div className="auth-content">
                                <p className="text-xs-center">REGISTER TO CONTINUE</p>
                                <form id="signup" method="POST">
                                    <div className="form-group">
                                        <label>Email</label>
                                        <input type="text" className="form-control underlined" name="email" id="email" placeholder="Enter email address" />
                                    </div>

                                    <div className="form-group">
                                        <label>Password</label>
                                        <input type="password" className="form-control underlined" name="password" id="password" placeholder="Enter password" />
                                    </div>

                                    <div className="form-group">
                                        <label>Confirm Password</label>
                                        <input type="password" className="form-control underlined" name="confirm_password" id="confirm_password" placeholder="Enter password" />
                                    </div>
                                    <div className="form-group">
                                        <button type="submit" className="btn btn-block btn-primary">Sign Up</button>
                                    </div>

                                    <div className="form-group">
                                        <p className="text-muted text-xs-center">Already have an account? <Link to="/login">Login</Link>
                                        </p>

                                    </div>
                                </form>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default Register;