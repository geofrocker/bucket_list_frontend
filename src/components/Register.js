import React, {Component} from 'react';
import {Link, Redirect} from 'react-router-dom';
import axios from 'axios';
import swal from 'sweetalert';


class Register extends Component{

    /**
     * This component is used to Register a user.
     * It handles the following methods:
     *  handleEmailChange() --> Used to set the state that is being changed in the value of the input to the state
     *  handlePasswordChange() --> Used to set the state of the password that is being changed
     *  handlePasswordChange1() --> used to change the state of the confirm password
     *  handleSubmit() --> Used to submit data to the specified endpoint
     *  render --> Used to return jsx
     */

    constructor(props) {
        super(props);
        this.state = {email: '', password: '', confirm_password: '', redirect: false};
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleEmailChange = this.handleEmailChange.bind(this);
        this.handlePasswordChange = this.handlePasswordChange.bind(this);
        this.handlePasswordChange1 = this.handlePasswordChange1.bind(this);

    }
    handleEmailChange(event){
        this.setState({email: event.target.value})
    }

    handlePasswordChange(event) {
        this.setState({password: event.target.value})
    }
    handlePasswordChange1(event) {
        this.setState({confirm_password: event.target.value})
    }

    handleSubmit(event){
        event.preventDefault();
        axios({
            url : 'https://ridge-bucket-list-api.herokuapp.com/api/v1/auth/register',
            data: {email: this.state.email, password: this.state.password, confirm_password:this.state.confirm_password},
            datatype: "json",
            method: "post"
        })
            .then((response)=>{
                swal("Success", response.data.success, "success");
                window.localStorage.setItem('token', response.data.token);
                this.setState({redirect:true});
            })
            .catch((xhr) =>{
                console.log(JSON.stringify(xhr));
                swal("Error!", xhr.response.data.error, 'error');

            });
    }
    render(){
        if (this.state.redirect){
            return <Redirect to="/bucketlists/create"/>
        }

        return (
            <div>
                <div className="auth">
                    <div className="auth-container">
                        <div className="card">
                            <header className="auth-header">
                                <h1 className="auth-title">
                                    BUCKET-LIST
                                </h1>
                            </header>
                            <div className="auth-content">
                                <p className="text-xs-center">REGISTER TO CONTINUE</p>
                                <form id="signup" method="POST" onSubmit={this.handleSubmit}>
                                    <div className="form-group">
                                        <label>Email</label>
                                        <input type="text" className="form-control underlined" value={this.state.email} onChange={this.handleEmailChange} name="email" id="email" placeholder="Enter email address" />
                                    </div>

                                    <div className="form-group">
                                        <label>Password</label>
                                        <input type="password" className="form-control underlined" value={this.state.password} onChange={this.handlePasswordChange} name="password" id="password" placeholder="Enter password" />
                                    </div>

                                    <div className="form-group">
                                        <label>Confirm Password</label>
                                        <input type="password" className="form-control underlined" value={this.state.password1} onChange={this.handlePasswordChange1} name="confirm_password" id="confirm_password" placeholder="Enter password" />
                                    </div>
                                    <div className="form-group">
                                        <button type="submit" className="btn btn-block btn-primary">Sign Up</button>
                                    </div>

                                    <div className="form-group">
                                        <p className="text-muted text-xs-center">Already have an account? <Link to="/login">Login</Link>
                                        </p>

                                    </div>
                                </form>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default Register;