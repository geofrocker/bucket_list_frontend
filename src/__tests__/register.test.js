import React from 'react';
import Register from '../components/Register';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import {MemoryRouter} from 'react-router-dom';

describe('<Register />', ()=>{
    it('contains signup form', function () {
        const wrapper = mount(
            <MemoryRouter>
                <Register/>
            </MemoryRouter>);
        const preventDefault = jest.fn();

        wrapper.setState({email:'ridgekimani@gmail.com', password:"milimani", confirm_password:"milimani"});
        expect(wrapper.find("#signup").length === 1);
        wrapper.find("#signup").simulate('submit', {preventDefault});
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(preventDefault).toBeCalled();
    });

    it('should change the values to the email input in the form', function(){
        const wrapper = mount(
            <MemoryRouter>
                <Register/>
            </MemoryRouter>);
        wrapper.find("#email").simulate('change', {target: {value: 'example@email.com'}});
        expect(toJson(wrapper)).toMatchSnapshot();

    });

    it('should change the values to the password input in the form', function(){
        const wrapper = mount(
            <MemoryRouter>
                <Register/>
            </MemoryRouter>);
        wrapper.find("#password").simulate('change', {target: {value: 'milimani2'}});
        expect(toJson(wrapper)).toMatchSnapshot();

    });

    it('should change the values to the password input', function(){
        const wrapper = mount(
            <MemoryRouter>
                <Register/>
            </MemoryRouter>);
        wrapper.find("#confirm_password").simulate('change', {target: {value: 'milimani2'}});
        expect(toJson(wrapper)).toMatchSnapshot();

    })

});
