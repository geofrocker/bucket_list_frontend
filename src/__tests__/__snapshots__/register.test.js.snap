// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> contains signup form 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "mc0ln4",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "mc0ln4",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Register>
      <div>
        <div
          className="auth"
        >
          <div
            className="auth-container"
          >
            <div
              className="card"
            >
              <header
                className="auth-header"
              >
                <h1
                  className="auth-title"
                >
                  BUCKET-LIST
                </h1>
              </header>
              <div
                className="auth-content"
              >
                <p
                  className="text-xs-center"
                >
                  REGISTER TO CONTINUE
                </p>
                <form
                  id="signup"
                  method="POST"
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <label>
                      Email
                    </label>
                    <input
                      className="form-control underlined"
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="Enter email address"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label>
                      Password
                    </label>
                    <input
                      className="form-control underlined"
                      id="password"
                      name="password"
                      onChange={[Function]}
                      placeholder="Enter password"
                      type="password"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label>
                      Confirm Password
                    </label>
                    <input
                      className="form-control underlined"
                      id="confirm_password"
                      name="confirm_password"
                      onChange={[Function]}
                      placeholder="Enter password"
                      type="password"
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <button
                      className="btn btn-block btn-primary"
                      type="submit"
                    >
                      Sign Up
                    </button>
                  </div>
                  <div
                    className="form-group"
                  >
                    <p
                      className="text-muted text-xs-center"
                    >
                      Already have an account? 
                      <Link
                        replace={false}
                        to="/login"
                      >
                        <a
                          href="/login"
                          onClick={[Function]}
                        >
                          Login
                        </a>
                      </Link>
                    </p>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </Register>
  </Router>
</MemoryRouter>
`;

exports[`<Register /> should change the values to the email input in the form 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "53q0jp",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "53q0jp",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Register>
      <div>
        <div
          className="auth"
        >
          <div
            className="auth-container"
          >
            <div
              className="card"
            >
              <header
                className="auth-header"
              >
                <h1
                  className="auth-title"
                >
                  BUCKET-LIST
                </h1>
              </header>
              <div
                className="auth-content"
              >
                <p
                  className="text-xs-center"
                >
                  REGISTER TO CONTINUE
                </p>
                <form
                  id="signup"
                  method="POST"
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <label>
                      Email
                    </label>
                    <input
                      className="form-control underlined"
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="Enter email address"
                      type="text"
                      value="example@email.com"
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label>
                      Password
                    </label>
                    <input
                      className="form-control underlined"
                      id="password"
                      name="password"
                      onChange={[Function]}
                      placeholder="Enter password"
                      type="password"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label>
                      Confirm Password
                    </label>
                    <input
                      className="form-control underlined"
                      id="confirm_password"
                      name="confirm_password"
                      onChange={[Function]}
                      placeholder="Enter password"
                      type="password"
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <button
                      className="btn btn-block btn-primary"
                      type="submit"
                    >
                      Sign Up
                    </button>
                  </div>
                  <div
                    className="form-group"
                  >
                    <p
                      className="text-muted text-xs-center"
                    >
                      Already have an account? 
                      <Link
                        replace={false}
                        to="/login"
                      >
                        <a
                          href="/login"
                          onClick={[Function]}
                        >
                          Login
                        </a>
                      </Link>
                    </p>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </Register>
  </Router>
</MemoryRouter>
`;

exports[`<Register /> should change the values to the password input 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "ig64qi",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "ig64qi",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Register>
      <div>
        <div
          className="auth"
        >
          <div
            className="auth-container"
          >
            <div
              className="card"
            >
              <header
                className="auth-header"
              >
                <h1
                  className="auth-title"
                >
                  BUCKET-LIST
                </h1>
              </header>
              <div
                className="auth-content"
              >
                <p
                  className="text-xs-center"
                >
                  REGISTER TO CONTINUE
                </p>
                <form
                  id="signup"
                  method="POST"
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <label>
                      Email
                    </label>
                    <input
                      className="form-control underlined"
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="Enter email address"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label>
                      Password
                    </label>
                    <input
                      className="form-control underlined"
                      id="password"
                      name="password"
                      onChange={[Function]}
                      placeholder="Enter password"
                      type="password"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label>
                      Confirm Password
                    </label>
                    <input
                      className="form-control underlined"
                      id="confirm_password"
                      name="confirm_password"
                      onChange={[Function]}
                      placeholder="Enter password"
                      type="password"
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <button
                      className="btn btn-block btn-primary"
                      type="submit"
                    >
                      Sign Up
                    </button>
                  </div>
                  <div
                    className="form-group"
                  >
                    <p
                      className="text-muted text-xs-center"
                    >
                      Already have an account? 
                      <Link
                        replace={false}
                        to="/login"
                      >
                        <a
                          href="/login"
                          onClick={[Function]}
                        >
                          Login
                        </a>
                      </Link>
                    </p>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </Register>
  </Router>
</MemoryRouter>
`;

exports[`<Register /> should change the values to the password input in the form 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "60e431",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "60e431",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Register>
      <div>
        <div
          className="auth"
        >
          <div
            className="auth-container"
          >
            <div
              className="card"
            >
              <header
                className="auth-header"
              >
                <h1
                  className="auth-title"
                >
                  BUCKET-LIST
                </h1>
              </header>
              <div
                className="auth-content"
              >
                <p
                  className="text-xs-center"
                >
                  REGISTER TO CONTINUE
                </p>
                <form
                  id="signup"
                  method="POST"
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <label>
                      Email
                    </label>
                    <input
                      className="form-control underlined"
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="Enter email address"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label>
                      Password
                    </label>
                    <input
                      className="form-control underlined"
                      id="password"
                      name="password"
                      onChange={[Function]}
                      placeholder="Enter password"
                      type="password"
                      value="milimani2"
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label>
                      Confirm Password
                    </label>
                    <input
                      className="form-control underlined"
                      id="confirm_password"
                      name="confirm_password"
                      onChange={[Function]}
                      placeholder="Enter password"
                      type="password"
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <button
                      className="btn btn-block btn-primary"
                      type="submit"
                    >
                      Sign Up
                    </button>
                  </div>
                  <div
                    className="form-group"
                  >
                    <p
                      className="text-muted text-xs-center"
                    >
                      Already have an account? 
                      <Link
                        replace={false}
                        to="/login"
                      >
                        <a
                          href="/login"
                          onClick={[Function]}
                        >
                          Login
                        </a>
                      </Link>
                    </p>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </Register>
  </Router>
</MemoryRouter>
`;
